doc/Command-Reference.md

```diff
@@ -7961,8 +7962,305 @@ Go Back To [Beginning of the document](#) or [Beginning of this section](#waterm
+  ```
+
+**sonic-package-manager upgrade**
+
+This command upgrades package on SONiC host to a newer version. The procedure of upgrading a package will restart the corresponding service. *NOTE*: this command requires elevated (root) privileges to run.
+
+- Usage:
+  ```
+  Usage: sonic-package-manager upgrade [OPTIONS] [PACKAGE_EXPR]
+
+    Upgrade package using [PACKAGE_EXPR] in format "<name>==<version>"
+
+  Options:
+    --from-repository TEXT  Fetch package directly from image registry
+                            repository NOTE: This argument is mutually exclusive
+                            with arguments: [package_expr, from_tarball].
+    --from-tarball FILE     Fetch package from saved image tarball NOTE: This
+                            argument is mutually exclusive with arguments:
+                            [from_repository, package_expr].
+    -f, --force             Force operation by ignoring failures
+    -y, --yes               Automatically answer yes on prompts
+    -v, --verbosity LVL     Either CRITICAL, ERROR, WARNING, INFO or DEBUG
+    --skip-host-plugins     Do not install host OS plugins provided by the
+                            package (CLI, etc). NOTE: In case when package host
+                            OS plugins are set as mandatory in package manifest
+                            this option will fail the installation.
+    --help                  Show this message and exit.
+  ```
+- Example:
+  ```
+  admin@sonic:~$ sudo sonic-package-manager upgrade dhcp-relay==2.0.0
+  ```
+  ```
+  admin@sonic:~$ sudo sonic-package-manager upgrade --from-repository azure/sonic-cpu-report:latest
+  ```
+  ```
+  admin@sonic:~$ sudo sonic-package-manager upgrade --from-tarball sonic-docker-image.gz
+  ```
+
+**sonic-package-manager reset**
+
+This comamnd resets the package by reinstalling it to its default version.
+
+- Usage:
+  ```
+  Usage: sonic-package-manager reset [OPTIONS] NAME
+
+    Reset package to the default version
+
+  Options:
+    -f, --force          Force operation by ignoring failures
+    -y, --yes            Automatically answer yes on prompts
+    -v, --verbosity LVL  Either CRITICAL, ERROR, WARNING, INFO or DEBUG
+    --skip-host-plugins  Do not install host OS plugins provided by the package
+                        (CLI, etc). NOTE: In case when package host OS plugins
+                        are set as mandatory in package manifest this option
+                        will fail the installation.
+    --help               Show this message and exit.
+  ```
+- Example:
+  ```
+  admin@sonic:~$ sudo sonic-package-manager reset dhcp-relay
+  ```
+
+**sonic-package-manager show package versions**
+
+This command will access repository for corresponding package and retrieve a list of available versions.
+
+- Usage:
+  ```
+  Usage: sonic-package-manager show package versions [OPTIONS] NAME
+
+    Print available versions
+
+  Options:
+    --all    Show all available tags in repository
```

Comment:
```suggestion
    --all    Show all versions available in the repository
```

Reply:
Is "--all" the default ? If not, what is the default ?

Reply:
No, it is not. --all is an option so it is not a default.